source/code/plugins/VMInsightsDataCollector.rb

```diff
@@ -99,17 +99,12 @@ def get_number_of_cpus
         def get_filesystems
             result = []
             df = File.join(@root, "bin", "df")
-            if @root != '/'
-                cmd = "#{df} --block-size=1 -T"
-            else
-                cmd = "#{df} --block-size=1 -T | awk '{print $2, $1, $7, $3, $5}'"
-            end
-            IO.popen(cmd, { :in => :close }) { |io|
+            IO.popen([df, "--block-size=1", "-T"], { :in => :close, :err => File::NULL }) { |io|
                 while (line = io.gets)
-                    if (line =~ /ext[234]/)
+                    a = line.split(" ")
+                    if (a[1] =~ /ext[234]/)
```

Comment:
@mthompson83  reworked how the file system type was being checked. now only checks the correct field for filesystem type thereby reducing the need for a separate unit test